controllers/volumereplication_controller.go

```diff
@@ -227,8 +227,8 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	return ctrl.Result{}, nil
 }
 
-func (r *VolumeReplicationReconciler) updateReplicationStatus(instance *replicationv1alpha1.VolumeReplication, state replicationv1alpha1.State, message string) error {
-	instance.Status.State = state
+func (r *VolumeReplicationReconciler) updateReplicationStatus(instance *replicationv1alpha1.VolumeReplication, message string) error {
```

Comment:
It will update the state when this will get called for the first time in reconcile. This should not be the case IMO. We should update the state only when we are successfully able to achieve/change the state.

Reply:
its ok to update the `state` when we reconcile for the first time. `Status.State` represents what's the last operation user performed on the CR. The conditions will reflect whether we are successfully able to achieve/change the state.